### PR TITLE
Feat show queue resources on job form

### PR DESCRIPTION
**Pull request by brnovasco**
- create bash template script for combining information from slurm about user available queues and resource limits an print results to stdout as a json;
- create router procedure for running a remote ssh command with the script created setting variables with the procedure input and returning the necessary information for the route client;
- create zod schema for parsing the script output;
- change the job form for showing the queue resources in the field for queue selection;

Screenshot from 2024-07-24 14-06-10

### CODE REVIEW

File: src/lib/schemas/user-partitions-response.ts [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+
+const cpusStateSchema = z.object({
+  allocated: z.string(),
+  idle: z.string(),
+  other: z.string(),
+  total: z.string(),
+});
+
+const groupQoSLimitSchema = z.object({
+  cpu: z.union([z.string(), z.null()]).optional(),
```
Comment (matyson) [inline src/lib/schemas/user-partitions-response.ts:11]: does it need to be an union?

File: src/server/api/routers/job.ts [hunk 2]
```diff
@@ -269,6 +276,85 @@ export const jobRouter = createTRPCRouter({
...
+    }
+
+    const parsed = userPartitionsResponseSchema.safeParse(
+      JSON.parse(stdout.trim()),
+    );
+    if (parsed.error) {
+      throw new Error(parsed.error.message);
+    }
+
+    function parseNullOrUndefined(value: string | null | undefined) {
```
Comment (matyson) [inline src/server/api/routers/job.ts:313]: unused function

File: src/server/api/routers/job.ts [hunk 3]
```diff
@@ -305,3 +391,5 @@ export const jobRouter = createTRPCRouter({
     return { jobs };
   }),
 });
+
+// echo $(sacctmgr show association -P -r format=Partition Users=bruno.carlos --noheader) | tr ' ' '|'
```
Comment (matyson) [inline src/server/api/routers/job.ts:395]: WTF?!

**Review by matyson — CHANGES_REQUESTED** [review]

File: src/lib/schemas/user-partitions-response.ts (hunk 1 above)
Comment (brnovasco) [reply src/lib/schemas/user-partitions-response.ts:11]: Not _really_! Thanks!

File: src/server/api/routers/job.ts (hunk 3 above)
Comment (brnovasco) [reply src/server/api/routers/job.ts:395]: At least It wasn't a typo! Fixing it!

File: src/server/api/routers/job.ts (hunk 2 above)
Comment (brnovasco) [reply src/server/api/routers/job.ts:313]: Fixing it!

**Review by matyson — APPROVED** [review]
looks good, thanks

**Pull request merged**